aws/session/shared_config.go

```diff
@@ -131,6 +135,14 @@ func (cfg *sharedConfig) setAssumeRoleSource(origProfile string, files []sharedC
 	if cfg.AssumeRole.SourceProfile == origProfile {
 		assumeRoleSrc = *cfg
 		assumeRoleSrc.AssumeRole = assumeRoleConfig{}
+	} else if cfg.AssumeRole.CredentialSource == "Ec2InstanceMetadata" {
```

Comment:
I think this would work equally well for `EcsContainer`, but I don't have a readily-available environment where I could test that.